### ISSUE TITLE
iSEEtree

### ISSUE THREAD

**Issue by RiboRings** [Bioconductor/Contributions#3451]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/microbiome/iSEEtree

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by RiboRings** [comment]
Tagging @kevinrue

**Comment by RiboRings** [comment]
Hi! I'm having a hard time connecting to the Bioconductor.org via ssh to push changes. I activated BiocCredentials account and checked that SSH is correct but still connection fails. Would you have any tip how to fix this issue?

```
> git remote -v
origin	git@github.com:microbiome/iSEEtree.git (fetch)
origin	git@github.com:microbiome/iSEEtree.git (push)
upstream	git@git.bioconductor.org:packages/iSEEtree.git (fetch)
upstream	git@git.bioconductor.org:packages/iSEEtree.git (push)

> git fetch --all
Fetching origin
Fetching upstream
ssh: connect to host git.bioconductor.org port 22: Operation timed out
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
error: could not fetch upstream
```

**Comment by kevinrue** [comment]
When did you set up the credentials and when did you test? 

It's been too long for me to remember but I suspect that Bioconductor has an automated script that runs on a schedule, probably daily, meaning that you may need to give it a bit of time.

Don't take my word for it, I'm genuinely trying to remember!

**Comment by RiboRings** [comment]
Thanks! I tested immediately after activating it, I'll try tomorrow.

**Comment by PeteHaitch** [comment]
Hi @RiboRings,

Thank you for submitting **iSEEtree** to Bioconductor.

I've completed my checklist review of **iSEEtree** and overall the package is in good shape and close to being ready for acceptance.
The main issue is in improving the vignette.
The vignette is the first thing many potential users will look at, so you want to explain when they might want to use **iSEEtree** and how it relates to other Bioconductor packages they may already be familiar with.

In my checklist review below I have separated the issues into **Required** and **Recommended** points that I would ask you to address before the package can be accepted.
Would you please provide line-by-line comments to my initial review so that I know what changes I'm looking for in my re-review.

Cheers,
Pete

----------

### Required

- [ ] The vignette must have an *Introduction* section. This should include motivation for inclusion to Bioconductor. 
  - [ ] Currently, the vignette doesn't really tell someone when they would want to use **iSEEtree** or how it differs from or extends **iSEE**. To my mind, you've got 2 sorts of users: (1) those familiar with microbiome packages (**mia**, **TreeSummarizedExperiment**, etc.) but not with **iSEE** and (2) users who are new to microbiome work but perhaps familiar with **iSEE** from other types of omics data. I think the vignette should point to the extensive iSEE documentation for new users in category (1) and the relevant documentation for new users in category (2). 
  - [ ] I come from category (2) and had no real idea what I was looking at when I read the vignette and launched the app, so its also important to some explanation of the new panels to the vignette.
- [ ] Installation instructions should be the officially supported Bioconductor instructions rather than `remotes::install_github()`-based ones.
- [ ] Man page for `iSEEtree::iSEE` should document `Usage`, `Arguments`, `Details`, `Value`, etc. like is done for `?iSEE::iSEE`. This might be achieved by cross-linking or otherwise re-using documentation from **iSEE** (see https://roxygen2.r-lib.org/articles/reuse.html)
- `R CMD build` reports `2 notes` on my system (see below):
  - [ ] I think the first is noting that a `LICENSE` file isn't required for a package with a Artistic-2.0 license (e.g., core Bioconductor packages with this license, such as https://github.com/Bioconductor/IRanges/, don't include that file).
  - [ ] For the second, have you asked the **iSEE** whether they'd consider exporting this function?

```r
  * checking top-level files ... NOTE
File
  LICENSE
  * checking dependencies in R code ... NOTE
Unexported object imported by a ':::' call: ‘iSEE:::.selectInputHidden’
  See the note in ?`:::` about the use of this operator.
```
- [ ] The runnable examples on each man page aren't always focused on the documented function. E.g., `?AbundanceDensityPlot` doesn't explicitly include that function anywhere in its example. I appreciate that some (if not all) of the exported functions are documenting **iSEE** panels that on their own may not be functional or useful, but perhaps there's a way to better tailor the examples to the specific function.

### Recommended

- [ ] Suggest running `spelling::spell_check_package()` because I spotted a few typos when reading the documentation.
- [ ] Not clear to me why these chunks are in the vignette. I suggest removing them.
  - [ ] `createVignette`chunk  isn't evaluated or echoed; why is it there?
  - [ ] `reproduce1` chunk is redundant because date is automatically added to rendered vignette by BiocStyle
  - [ ] `reproduce2` chunk and the `startTime <- Sys.time()` in `vignetteSetup` chunk: timing info isn't relevant to user
- [ ] `BiocCheck::BiocCheck()` recommends some biocViews terms that you might consider (e.g., `GUI, ShinyApps`)
- [ ] Good job adding unit tests. You could take a look at output of `covr::report()` to identify areas that may benefit from testing.

**Comment by RiboRings** [comment]
Hi! Thank you for the thorough feedback! I will work on those points hopefully in a couple of weeks.

Regarding .selectInputHidden, a PR is open in iSEE: https://github.com/iSEE/iSEE/pull/656

**Comment by kevinrue** [comment]
> Regarding .selectInputHidden, a PR is open in iSEE: https://github.com/iSEE/iSEE/pull/656

I completely forgot about that one 🥹

I think the PR covers the technical bit of exporting, but best practice still require a tiny comment about the export in the man page. I'm about to be offline for a week, and this summer is incredibly packed for me, but this export is part of a much larger need to export a whole series of methods needed by various extension packages. iSEE's API for extensions has been growing faster than we've been able to properly document it, so some of the methods are still internal when they should be exported (and renamed without a dot prefix)

**Comment by RiboRings** [comment]
Hi @PeteHaitch!

The package is ready for the next round of review. Below I will mention how the changes map to your feedback

### The vignette must have an Introduction section. This should include motivation for inclusion to Bioconductor.

Now the vignettes contain an [introduction section](https://microbiome.github.io/iSEEtree/#introduction) and a [description of the panels](https://microbiome.github.io/iSEEtree/#panels). New users are redirected to the main documentations of iSEE and mia.

### Installation instructions should be the officially supported Bioconductor instructions rather than remotes::install_github()-based ones.

Now both the readme and the get started page give Bioconductor installation instructions.

###  Man page for `iSEEtree::iSEE` should document Usage, Arguments, Details, Value, etc. like is done for `iSEE::iSEE`.

Now the [vignettes of iSEEtree::iSEE](https://microbiome.github.io/iSEEtree/reference/iSEE.html) inherit the sections from iSEE::iSEE.

### R CMD build reports 2 notes on my system

The LICENSE note was fixed. The .selectInputHidden note is in the agenda but will probably be solved at a later point depending on the iSEE schedule.

### The runnable examples on each man page aren't always focused on the documented function.

Now every vignette shows how to view the slots and start iSEE with initial custom panel specifically for the corresponding function. For example: https://microbiome.github.io/iSEEtree/reference/AbundanceDensityPlot.html

### Suggest running spelling::spell_check_package()

For me it didn't find any mistake.

### Not clear to me why these chunks are in the vignette

I removed them

### BiocCheck::BiocCheck() recommends some biocViews terms that you might consider

I added them. However, the recommendations are not always the same when I run the check.

### Good job adding unit tests

I added more. Some method like .createObservers are difficult to test because they require an interactive session. Others are tested but not all their parts are detected by codecov.

**Comment by PeteHaitch** [comment]
Thank you for responding to the review and making many of the requested changes, @RiboRings.

It seems like you might be confused about the difference between a vignette ([a file with Bioconductor-specific requirements](https://contributions.bioconductor.org/docs.html?q=vigne#vignettes)) and other documentation (e.g., `README.Rmd`
It looks like you've made your changes only to the `README.md` but not to the actual vignette (`vignettes/iSEEtree.Rmd`).
That is, the vignette still does not contain an **Introduction** section and still needs the below points from the initial review to be addressed:

- [ ] Not clear to me why these chunks are in the vignette. I suggest removing them.
  - [ ] `createVignette`chunk  isn't evaluated or echoed; why is it there?
  - [ ] `reproduce1` chunk is redundant because date is automatically added to rendered vignette by BiocStyle
  - [ ] `reproduce2` chunk and the `startTime <- Sys.time()` in `vignetteSetup` chunk: timing info isn't relevant to user

Please fix these.

`spelling::spell_check_package()` does give a bunch of false-positives but reals ones include:

```r
> spelling::spell_check_package()
compositional              README.md:22,90
contructor                 RDAPlot.Rd:9
funcionality               description:2
```

**Comment by RiboRings** [comment]
Thanks @PeteHaitch for clearing that out!

I moved the introduction to the vignette and reorganised it so that it's hopefully easier to follow.

I am getting this warning on Bioconductor server, not sure yet what's the reason:
`WARNING: Package listed as VignetteEngine or VignetteBuilder but
      not currently Suggested.`

I checked and both knitr and rmarkdown are under Suggested in DESCRIPTION.

**Comment by PeteHaitch** [comment]
Thank you for making those changes, @RiboRings.

I'm now happy to accept **iSEEtree** into Bioconductor.
Congratulations and thank you for your contribution!

Regarding the warning you are receiving from `BiocCheck::BiocCheck()` (`WARNING: Package listed as VignetteEngine or VignetteBuilder but not currently Suggested.`) - I can reproduce it but I don't understand it either.
I see another developer has also been recently confused by this (https://github.com/Bioconductor/Contributions/issues/3379#issuecomment-2244550550).
@LiNk-NY Could you please explain what's happening here? Is it a false positive?

**Comment by LiNk-NY** [comment]
> Could you please explain what's happening here? Is it a false positive?

Thanks @PeteHaitch , yes, it was a false positive. It should be fixed in version `1.41.7`.

**Comment by lshep** [comment]
The default branch of your GitHub repository has been added to Bioconductor's
git repository as branch devel.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/RiboRings.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("iSEEtree")`. The package 'landing page' will be created at

https://bioconductor.org/packages/iSEEtree

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
